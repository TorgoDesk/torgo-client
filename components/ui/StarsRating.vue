<template>
  <span>
    <i
      v-for="n in stars"
      :key="`stars_${n}`"
      class="fas fa-star fa-sm fa-fw text-warning"
    ></i
    ><i
      v-for="m in grayStars"
      :key="`grayStars_${m}`"
      class="fas fa-star fa-sm fa-fw text-gray-400"
    ></i>
  </span>
</template>

<script>
export default {
  props: {
    stars: Number,
  },
  data() {
    return {};
  },
  computed: {
    grayStars: function () {
      return 5 - this.stars;
    },
  },
};
</script>

<style>
</style>
