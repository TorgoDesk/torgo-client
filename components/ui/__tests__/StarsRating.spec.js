import { createLocalVue, mount } from "@vue/test-utils";
import StarsRating from "@/components/ui/StarsRating";

const factory = () => {
  return mount(StarsRating, {
    propsData: {
      stars: 4
    }
  })
}

describe('StarsRating', () => {
    const wrapper = factory()
    test("renders properly", () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
})
