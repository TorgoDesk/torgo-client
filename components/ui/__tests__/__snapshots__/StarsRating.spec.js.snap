// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StarsRating renders properly 1`] = `<span><i class="fas fa-star fa-sm fa-fw text-warning"></i><i class="fas fa-star fa-sm fa-fw text-warning"></i><i class="fas fa-star fa-sm fa-fw text-warning"></i><i class="fas fa-star fa-sm fa-fw text-warning"></i><i class="fas fa-star fa-sm fa-fw text-gray-400"></i></span>`;
